app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -138,6 +155,231 @@ public static void playOnBackgroundPlayer(final Context context, final PlayQueue
         startService(context, getPlayerIntent(context, BackgroundPlayer.class, queue, resumePlayback));
     }
 
+    public static void downloadPlaylist(final PlaylistFragment context, final PlayQueue queue) {
```

Comment:
Do all of these `PlaylistFragment` types always have to be like that? Couldn't a Fragment or Activity work?

Reply:
What exactly do you mean?

Reply:
What if one wants to download a playlist from a different fragment? It would be better if the type of the parameter `context` was the base class `Fragment` and not `PlaylistFragment`, so that the method can be used everywhere.

Reply:
Theoretically I can get away with using FragmentActivity instead (there would be some implications; like I'm passing onError implementation of PlaylistFragment for reactive stream failure, while in FragmentActivity won't be able to do that. If there were an abstract class, I could've used that instead, for this specific use case.
Apart from that, the initial call to initialize DownloadDialog#newInstance requires PlaylistFragment#callback now. Again, an abstract class would help here too, so in future you would need to extend that class wherever you would want to initialize this download dialog.
Right now it's tightly coupled with PlaylistFragment, but the change you're suggesting would require a lot of changes in many parts of your source.
I'll leave it upto you to decide for this.

Reply:
Mmhh OK, in case I ever need to use it from elsewhere I will implement it myself

Reply:
Good, in this case it is time for a last round of testing and then we should be ready, finally :)
Thanks again for keeping your branch updated @VishalNehra. I am sorry, we've kept you waiting way too long.